### ISSUE TITLE
Parameterized display and model annotation

### ISSUE THREAD

**Issue by alexis-** [WildGums/Orc.FilterBuilder#30]
Now that there is an option to convert filters to Linq Expressions, it makes sense to use them with ORM frameworks to run parameterized database queries. It was of course the motivation behind my PR.

Although, these aspects may be lacking :
- All operators available in Orc.FilterBuilder may not be implemented in the ORM/backing store engine.
- Some models may not have all their public properties implemented in the table, or simply shouldn't be filtered on these columns.
- Some properties may require to be displayed in a specific way (enums, int values, ...)

I will probably be working on addressing these issues soon. My questions are :
- Is this of any interest to you ?
- Is there any item I missed and you believe I should also tackle ?
- Do you have any tip or bits I could use before I begin ?

As the title suggests, I am leaning toward using annotations.
Possibly reusing some existing ones -- if compatible with PCL -- such as Browsable, etc. Although, this may restrict possibilities, and as such may not be such a good idea.

**Comment by GeertvanHorrik** [comment]
Yes, we are definitely interested in this. We have been working on Orc.Metadata which allows us to use adapters that don't require reflection / classes but could fetch the data directly from a database. It might be good to keep that in mind when working on this feature. Note that Orc.FilterBuilder can't directly reference any EF code. If we need additional logic for the ORM, we could have Orc.FilterBuilder.Providers.EntityFramework (or whatever ORM you are using).

I think the most important part for now is to finalize the ToLinqExpression PR so we have your current work merged before starting on a new PR (could result in too many things that need to be merged and can cause conflicts).
